activerecord/lib/active_record/insert_all.rb

```diff
@@ -34,11 +36,11 @@ def primary_keys
 
 
     def skip_duplicates?
-      on_duplicate == :skip
+      !@raise && !@upsert
```

Comment:
Doesn’t it add `ON CONFLICT DO NOTHING` (PostgreSQL) even we haven’t specified any options? Looks so.

We don’t need to add _on conflict_ clause in that case.

Shouldn’t it be:

```suggestion
      !@raise && (@upsert != false)
```

Reply:
You're thinking of `insert_all!`, `insert_all` (no-bang) skips by default.